test/unit/models/publishes_to_publishing_api_test.rb

```diff
@@ -2,85 +2,68 @@
-  teardown do
-    DatabaseCleaner.clean_with :truncation
-  end
+    def persisted?
+      true
+    end
 
-  test "generating a content_id" do
-    test_uuid = SecureRandom.uuid
-    SecureRandom.stubs(uuid: test_uuid)
-    organisation = create(:organisation)
-
-    assert organisation.valid?
-    assert_equal test_uuid, organisation.content_id
+    def search_link
+      "test_link"
+    end
   end
 
-  test "create publishes to Publishing API if not disallowed" do
-    organisation = build(:organisation)
-    Whitehall::PublishingApi.expects(:publish_async).with(organisation)
-    organisation.save
+  def include_module(object)
+    class << object
+      include PublishesToPublishingApi
+    end
+    object
   end
 
-  test "update publishes to Publishing API if not disallowed" do
-    organisation = create(:organisation)
-    Whitehall::PublishingApi.expects(:publish_async).with(organisation)
-    organisation.update_attribute(:name, 'Edited org')
+  setup do
+    TestObj.stubs(:after_commit).with(:publish_to_publishing_api,
+                                      {if: :can_publish_to_publishing_api?})
+    TestObj.stubs(:after_commit).with(:publish_gone_to_publishing_api,
+                                      {on: :destroy})
   end
 
-  test "create does not publish to Publishing API if disallowed" do
-    organisation = build(:organisation)
-    organisation.stubs(:can_publish_to_publishing_api?).returns(false)
-    Whitehall::PublishingApi.expects(:publish_async).never
-    organisation.save
+  test "it hooks up publish_to_publishing_api correctly" do
+    TestObj.expects(:after_commit).with(:publish_to_publishing_api,
+                                         {if: :can_publish_to_publishing_api?})
+    include_module(TestObj.new)
   end
 
-  test "update does not publish to Publishing API if disallowed" do
-    organisation = create(:organisation)
-    organisation.stubs(:can_publish_to_publishing_api?).returns(false)
-    Whitehall::PublishingApi.expects(:publish_async).never
-    organisation.update_attribute(:name, 'Edited org')
+  test "it hooks up publish_gone_to_publishing_api correctly" do
+    TestObj.expects(:after_commit).with(:publish_gone_to_publishing_api,
+                                         {on: :destroy})
+    include_module(TestObj.new)
   end
 
-  test "update publishes to Publishing API using :en locale when no translated fields are set" do
-    person = create(:person, attributes_for(:person).except(:biography))
-
-    content_item = PublishingApiPresenters.presenter_for(person)
-    requests = [
-      stub_publishing_api_put_content(content_item.content_id, content_item.content),
-      stub_publishing_api_put_links(content_item.content_id, links: content_item.links),
-      stub_publishing_api_publish(content_item.content_id, locale: content_item.content[:locale], update_type: 'major')
-    ]
-
-    assert_all_requested(requests)
+  test "can_publish_to_publishing_api_returns_true_when_object_persisted" do
```

Comment:
You've somehow snake cased your test name here plus elsewhere a few times in this file.